### PR TITLE
[Tier-2] CEPH-83575139 - Ensure a bucket only sync to specific zone

### PR DESCRIPTION
**Pull request by anrao19**
[Tier-2] CEPH-83575139 - Ensure a bucket only sync to specific zone

automated remaining semantic in : https://docs.google.com/spreadsheets/d/1-ZW8YF9CMQQdjTv3aobCAiqQa_-lYtb3WjbdoYEH7Ms/edit?usp=sharing

log: 
http://magna002.ceph.redhat.com/ceph-qe-logs/anuchaithra/test_multisite_granular_bucketsync_enable_enable.console.log
http://magna002.ceph.redhat.com/ceph-qe-logs/anuchaithra/test_multisite_granular_bucketsync_enabled_forbidden.console.log
http://magna002.ceph.redhat.com/ceph-qe-logs/anuchaithra/test_multisite_granular_bucketsync_forbidden_allowed.console.log
http://magna002.ceph.redhat.com/ceph-qe-logs/anuchaithra/test_multisite_granular_bucketsync_forbidden_enabled.console.log
http://magna002.ceph.redhat.com/ceph-qe-logs/anuchaithra/test_multisite_granular_bucketsync_forbidden_forbidden.console.log

### CODE REVIEW

**Review by ckulal — COMMENTED** [review]
lgtm

File: rgw/v2/tests/s3_swift/test_multisite_bucket_granular_sync_policy.py [hunk 1]
```diff
@@ -270,9 +278,92 @@ def test_exec(config, ssh_con):
...
+                            ):
+                                if bkt_objects != config.objects_count * 2:
+                                    raise TestExecError(
+                                        f"Object did not sync in bucket {bkt.name}, but found {bkt_objects}"
+                                    )
+                                log.info(
+                                    f"Object synced for bucket {bkt.name}, on another site as expected"
+                                )
+                            else:
+                                if bkt_objects == config.objects_count * 2:
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/test_multisite_bucket_granular_sync_policy.py:359]: @anrao19 , could you please change the condition to `if bkt_objects != config.objects_count:`
instead of particularly checking if equals to double of object count, this conditional covers the other corner cases as well right

File: rgw/v2/tests/s3_swift/test_multisite_bucket_granular_sync_policy.py [hunk 2]
```diff
@@ -270,9 +278,92 @@ def test_exec(config, ssh_con):
...
+                            )
+                            re_cmd_output = json.loads(stats_stdout.read().decode())
+                            log.info(f"re_cmd_output : {re_cmd_output}")
+                            new_object_count = num_objects + config.objects_count
+                            if (
+                                re_cmd_output["usage"]["rgw.main"]["num_objects"]
+                                != new_object_count
+                            ):
+                                raise TestExecError(
+                                    f"Failed to upload object to bucket {bkt.name}"
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/test_multisite_bucket_granular_sync_policy.py:332]: @anrao19 could you modify the exception message to "number of objects not updated in the bucket stats after new objects upload"

Comment (anrao19) [reply rgw/v2/tests/s3_swift/test_multisite_bucket_granular_sync_policy.py]: @hmaheswa changes made

**Review by hmaheswa — APPROVED** [review]

**Review by mkasturi18 — APPROVED** [review]
